x-pack/elastic-agent/pkg/agent/transpiler/vars_test.go

```diff
@@ -215,7 +215,8 @@ func TestVars_ReplaceWithProcessors(t *testing.T) {
 			},
 		},
 		"dynamic",
-		processers)
+		processers,
+		nil)
```

Comment:
Definitely need to add tests here. This already has good coverage for the current code, please add testing for the new fetching context providers as well.

Reply:
I can add tests once we verify the feature implementation so as to avoid going tuning during review cycles, wdyt?